tauri-macros/src/command.rs

```diff
@@ -0,0 +1,116 @@
+use proc_macro2::TokenStream;
+use quote::{format_ident, quote};
+use syn::{
+  parse::Parser, punctuated::Punctuated, FnArg, Ident, ItemFn, Meta, NestedMeta, Pat, Path, Token,
+  Type,
+};
+
+pub fn generate_command(attrs: Vec<NestedMeta>, function: ItemFn) -> TokenStream {
+  // Check if "webview" attr was passed to macro
+  let uses_webview = attrs.iter().any(|a| {
+    if let NestedMeta::Meta(Meta::Path(path)) = a {
+      path
+        .get_ident()
+        .map(|i| *i == "with_webview")
+        .unwrap_or(false)
+    } else {
+      false
+    }
+  });
+
+  let fn_name = function.sig.ident.clone();
+  let fn_wrapper = format_ident!("{}_wrapper", fn_name);
+
+  // Split function args into names and types
+  let (mut names, mut types): (Vec<Ident>, Vec<Path>) = function
+    .sig
+    .inputs
+    .iter()
+    .map(|param| {
+      let mut arg_name = None;
+      let mut arg_type = None;
+      if let FnArg::Typed(arg) = param {
+        if let Pat::Ident(ident) = arg.pat.as_ref() {
+          arg_name = Some(ident.ident.clone());
+        }
+        if let Type::Path(path) = arg.ty.as_ref() {
+          arg_type = Some(path.path.clone());
+        }
+      }
+      (
+        arg_name.clone().unwrap(),
+        arg_type.unwrap_or_else(|| panic!("Invalid type for arg \"{}\"", arg_name.unwrap())),
+      )
+    })
+    .unzip();
+
+  // If function doesn't take the webview, wrapper just takes webview generically and ignores it
+  // Otherwise the wrapper uses the specific type from the original function declaration
+  let mut webview_arg_type = quote!(tauri::WebviewManager<A>);
+  let mut application_ext_generic = quote!(<A: tauri::ApplicationExt>);
+  let webview_arg_maybe = match types.first() {
+    Some(first_type) if uses_webview => {
+      // Give wrapper specific type
+      webview_arg_type = quote!(#first_type);
+      // Generic is no longer needed
+      application_ext_generic = quote!();
+      // Remove webview arg from list so it isn't expected as arg from JS
+      types.drain(0..1);
+      names.drain(0..1);
+      // Tell wrapper to pass webview to original function
+      quote!(_webview,)
+    }
+    // Tell wrapper not to pass webview to original function
+    _ => quote!(),
+  };
+
+  quote! {
+    #function
+    pub fn #fn_wrapper #application_ext_generic(_webview: #webview_arg_type, arg: serde_json::Value) -> Option<tauri::InvokeResponse> {
+      #[derive(serde::Deserialize)]
```

Comment:
```suggestion
      #[derive(::serde::Deserialize)]
```
This one might require more thought because idk how this will handle `serde` not having the `"derive"` feature enabled - or what the solution might be